### PR TITLE
[REVIEW] Fix an error in IO with `GzipFile` type

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description

Fixes: #10590 

This PR fixes an issue where the file-like object doesn't have a `size` attribute, we will manually compute the size of the file.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/utils/ioutils.py [hunk 1]
```diff
@@ -1953,7 +1953,15 @@ def _fsspec_data_transfer(
 
     # Calculate total file size
     if file_like:
-        file_size = path_or_fob.size
+        try:
+            file_size = path_or_fob.size
+        except AttributeError:
+            # Find file size of there is no `size`
+            # attribute
+            old_file_position = path_or_fob.tell()
```
Comment (vuule) [inline python/cudf/cudf/utils/ioutils.py:1961]: Could we have a test where `old_file_position != 0`?

Comment (galipremsagar) [reply python/cudf/cudf/utils/ioutils.py:1961]: Done

File: python/cudf/cudf/utils/ioutils.py [hunk 2]
```diff
@@ -1953,7 +1953,15 @@ def _fsspec_data_transfer(
 
     # Calculate total file size
     if file_like:
-        file_size = path_or_fob.size
+        try:
+            file_size = path_or_fob.size
+        except AttributeError:
+            # Find file size of there is no `size`
```
Comment (brandon-b-miller) [inline python/cudf/cudf/utils/ioutils.py:1959]:
```suggestion
            # Find file size if there is no `size`
```

**Review by brandon-b-miller — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**